pyhf/__init__.py

```diff
@@ -144,10 +144,10 @@ def add_or_get_modifier(self, channel, sample, modifier_def):
         """
         # get modifier class associated with modifier type
         try:
-            modifier_cls = modifiers.registry.get(modifier_def['type'])
+            modifier_cls = modifiers.registry[modifier_def['type']]
         except KeyError:
-            raise RuntimeError('Modifier type not implemented yet (processing {0:s}). Current modifier types: {1}'.format(modifier_def['type'], modifiers.registry.keys()))
-
+            log.exception('Modifier type not implemented yet (processing {0:s}). Current modifier types: {1}'.format(modifier_def['type'], modifiers.registry.keys()))
+            raise
```

Comment:
This should just be `raise modifiers.InvalidModifier` since we have that [error class available](https://github.com/diana-hep/pyhf/blob/master/pyhf/modifiers/__init__.py#L7-L8).

Reply:
done